static/js/blame.js

```diff
@@ -156,22 +156,42 @@ var BlameStripHoverHandler = new (class BlameStripHoverHandler {
     // element will null this out as a means of letting the user get rid of the
     // popup if they didn't actually want to see the pop-up.
     this.mouseElement = null;
+    // Set to true if the user clicks on the blame strip. This will keep the
+    // keep the popup visible until the user clicks elsewhere.
+    this.keepVisible = false;
 
     for (let element of document.querySelectorAll(".blame-strip")) {
       element.addEventListener("mouseenter", this);
       element.addEventListener("mouseleave", this);
-      element.addEventListener("click", this);
     }
 
     BlamePopup.popup.addEventListener("mouseenter", this);
     BlamePopup.popup.addEventListener("mouseleave", this);
+    window.addEventListener("click", this, {capture: true});
+    document.getElementById("scrolling").addEventListener("scroll", this, {capture: true});
```

Comment:
`passive: true`?

Reply:
Also why does this need to be capturing? Same for the scroll listener.

Reply:
> `passive: true`?

I don't think passive does anything for scroll events. At least APZ doesn't care about whether scroll listeners are passive or not. Maybe I'm mistaken?

Reply:
> Also why does this need to be capturing? Same for the scroll listener.

The click needs to be capturing because if you e.g. click on a code fragment that produces a context menu, it seems to consume the click and then we end up with both the blame popup and the context menu. Having a capture listener here ensures we get the click even in that case. I can add a comment to that effect. The scroll listener probably doesn't need to be capturing, I can drop that.

Reply:
> I don't think passive does anything for scroll events. At least APZ doesn't care about whether scroll listeners are passive or not. Maybe I'm mistaken?

Ah, yeah, maybe I was thinking of wheel events, nvm.

> The click needs to be capturing because if you e.g. click on a code fragment that produces a context menu, it seems to consume the click and then we end up with both the blame popup and the context menu. Having a capture listener here ensures we get the click even in that case. I can add a comment to that effect. The scroll listener probably doesn't need to be capturing, I can drop that.

Ah, so this is so that we don't hit the `event.stopPropagation()` in the context-menu code? That should probably be changed to something else... And actually given now we don't always recreate the menu it may not be needed at all anymore.

Reply:
I changed /capture/passive/ for the scroll listener. The `passive` won't help but it won't hurt either, so whatever.

For the click - I guess it's the stopPropagation() in the context-menu code, but I didn't actually verify. In general it seems like for things like this a capturing listener seems more robust.

Reply:
It seems more reliable for hiding the popup, but not for showing it, right? Anyhow nbd.

Reply:
> It seems more reliable for hiding the popup, but not for showing it, right? Anyhow nbd.

Ah, good point. I was only considering the hiding scenario at the time but you're right it probably is not as good for the showing scenario.